worlds/timespinner/__init__.py

```diff
@@ -141,6 +143,76 @@ def fill_slot_data(self) -> Dict[str, object]:
+        self.options.damage_rando_overrides.value = slot_data["DamageRandoOverrides"]
+        self.options.hp_cap.value = slot_data["HpCap"]
+        self.options.level_cap.value = slot_data["LevelCap"]
+        self.options.extra_earrings_xp.value = slot_data["ExtraEarringsXP"]
+        self.options.boss_healing.value = slot_data["BossHealing"]
+        self.options.shop_fill.value = slot_data["ShopFill"]
+        self.options.shop_warp_shards.value = slot_data["ShopWarpShards"]
+        self.options.shop_multiplier.value = slot_data["ShopMultiplier"]
+        self.options.loot_pool.value = slot_data["LootPool"]
+        self.options.drop_rate_category.value = slot_data["DropRateCategory"]
+        self.options.fixed_drop_rate.value = slot_data["FixedDropRate"]
+        self.options.loot_tier_distro.value = slot_data["LootTierDistro"]
+        self.options.show_bestiary.value = slot_data["ShowBestiary"]
+        self.options.show_drops.value = slot_data["ShowDrops"]
+        self.options.enter_sandman.value = slot_data["EnterSandman"]
+        self.options.dad_percent.value = slot_data["DadPercent"]
+        self.options.rising_tides.value = slot_data["RisingTides"]
+        self.options.unchained_keys.value = slot_data["UnchainedKeys"]
+        self.options.back_to_the_future.value = slot_data["PresentAccessWithWheelAndSpindle"]
+        self.options.traps.value = slot_data["Traps"]
+        self.options.death_link.value = slot_data["DeathLink"]
+        # Readonly slot_data["StinkyMaw"]
+        # data
+        # Readonly slot_data["PersonalItems"]
+        self.precalculated_weights.pyramid_keys_unlock = slot_data["PyramidKeysGate"]
+        self.precalculated_weights.present_key_unlock = slot_data["PresentGate"]
+        self.precalculated_weights.past_key_unlock = slot_data["PastGate"]
+        self.precalculated_weights.time_key_unlock = slot_data["TimeGate"]
+        # rising tides
+        if (slot_data["Basement"] > 1):
+            self.precalculated_weights.flood_basement = True
+        if (slot_data["Basement"] == 2):
+            self.precalculated_weights.flood_basement_high = True
+        self.precalculated_weights.flood_xarion = slot_data["Xarion"]
+        self.precalculated_weights.flood_maw = slot_data["Maw"]
+        self.precalculated_weights.flood_pyramid_shaft = slot_data["PyramidShaft"]
+        self.precalculated_weights.flood_pyramid_back = slot_data["BackPyramid"]
+        self.precalculated_weights.flood_moat = slot_data["CastleMoat"]
+        self.precalculated_weights.flood_courtyard = slot_data["CastleCourtyard"]
+        self.precalculated_weights.flood_lake_desolation = slot_data["LakeDesolation"]
+        self.precalculated_weights.flood_lake_serene = not slot_data["DryLakeSerene"]
+        self.precalculated_weights.flood_lake_serene_bridge = slot_data["LakeSereneBridge"]
+        self.precalculated_weights.flood_lab = slot_data["Lab"]
+
+        return slot_data
```

Comment:
can probably move this up to the top because if you are passed in slot_data then you just want to return it and the rest of the code is meaningless
alternativly break this up into two methods, one for checking for and applying slot data and one (interpret_slot_data) to blindly return the slot data it gets passed by UT

Reply:
i though about splitting it, but then how do i call that? `actually_interpret_slot_data`? 
i mean a method named `interpret_slot_data` this is what you expect it todo, interpret the slot data and alter state

The who side effect it has that if you return a value you get an extra property on the multiworld is an akward api and i rather keep that nicely localy to this method and spread it further in my code

Reply:
that's why i suggested moving the return up to the top (if slot_data) instead if you don't want to break it up into multiple methods :p

Reply:
i understood

Reply:
The method recieves a `slot_data` dict or nothing, if its nothing it will try to obtain one from `re_gen_passthrough`
and it will use the `slot_data` it had to update the state

fast forwarding out of a the method if it does receive a slot_data makes no sense for the name of the method
splitting it up makes no sense either as the method that acutallly applied the slot_data should be named `interpret_slot_data`

Therefor the UT api is messy, but this works and i am happy with it

Reply:
> The only downside to this method is that on the first pass of the UT it will update a state that is getting thrown away be a regen, but i am ok with that

this was the only thing i was talking about, but I agree that the API is messy (and may go through some changes soon:tm:)
regardless it should function fine